# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-hal/src/vulkan/device.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/wgpu-hal/src/vulkan/device.rs
--- a/third_party/rust/wgpu-hal/src/vulkan/device.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/wgpu-hal/src/vulkan/device.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -4,6 +4,7 @@
     mem::{self, MaybeUninit},
     num::NonZeroU32,
     ptr,
+    time::Duration,
 };
 
 use arrayvec::ArrayVec;
@@ -895,7 +896,7 @@
                     if let Some(ref debug) = naga_shader.debug_source {
                         temp_options.debug_info = Some(naga::back::spv::DebugInfo {
                             source_code: &debug.source_code,
-                            file_name: debug.file_name.as_ref().into(),
+                            file_name: debug.file_name.as_ref(),
                             language: naga::back::spv::SourceLanguage::WGSL,
                         })
                     }
@@ -1931,7 +1932,7 @@
                         .as_ref()
                         .map(|d| naga::back::spv::DebugInfo {
                             source_code: d.source_code.as_ref(),
-                            file_name: d.file_name.as_ref().into(),
+                            file_name: d.file_name.as_ref(),
                             language: naga::back::spv::SourceLanguage::WGSL,
                         });
                 if !desc.runtime_checks.bounds_checks {
@@ -2443,9 +2444,12 @@
         &self,
         fence: &super::Fence,
         wait_value: crate::FenceValue,
-        timeout_ms: u32,
+        timeout: Option<Duration>,
     ) -> Result<bool, crate::DeviceError> {
-        let timeout_ns = timeout_ms as u64 * super::MILLIS_TO_NANOS;
+        let timeout_ns = timeout
+            .unwrap_or(Duration::MAX)
+            .as_nanos()
+            .min(u64::MAX as _) as u64;
         self.shared.wait_for_fence(fence, wait_value, timeout_ns)
     }
 